nanome_rmsd/rmsd_menu.py

```diff
@@ -18,47 +20,160 @@ def _request_refresh(self):
+            self.rmsd_score_label.text_value = "--"
+        else:
+            self.rmsd_score_label.text_value = str("%.3g"%value)
+        self._plugin.update_content(self.rmsd_score_label)
 
     def make_plugin_usable(self, state = True):
         self._run_button.unusable = not state
         self._plugin.update_button(self._run_button)
 
     # change the complex list
     def change_complex_list(self, complex_list):
-
+        
+        # a button in the receptor list is pressed
         def mobile_pressed(button):
             if self._selected_mobile != None:
-                self._selected_mobile.selected = False
-            button.selected = True
-            self._selected_mobile = button
-            self.receptor_text.text_value ="Receptor: "+ button.complex.name
-            self._plugin.update_menu(self._menu)
-
+                self._selected_mobile.selected = False 
+                if self._selected_mobile.complex != button.complex: 
+                    button.selected = True
+                    self._selected_mobile = button
+                    self.receptor_text.text_value ="Receptor: "+ button.complex.name
+                else: 
+                    self._selected_mobile = None
+                    self.receptor_text.text_value = "Receptor: Unselected"
+            else: 
+                button.selected = True
+                self._selected_mobile = button
+                self.receptor_text.text_value ="Receptor: "+ button.complex.name
+            if not self.change_error("unselected"):
+                if not self.change_error("select_same"):
+                    if not self.change_error("different_size"):
+                        self.change_error("different_order")
+            self._plugin.update_content(self._show_list)
+            self._plugin.update_content(self.receptor_text)
+            self._plugin.update_content(self.target_text)
+
+        # a button in the target list is pressed
         def target_pressed(button):
             if self._selected_target != None:
-                self._selected_target.selected = False
-            button.selected = True
-            self._selected_target = button
-            self.target_text.text_value = "Target: "+button.complex.name
-            self._plugin.update_menu(self._menu)
+                self._selected_target.selected = False 
+                if self._selected_target.complex != button.complex: 
+                    button.selected = True
+                    self._selected_target = button
+                    self.target_text.text_value ="Target: "+ button.complex.name
+                else: 
+                    self._selected_target = None
+                    self.target_text.text_value = "Target: Unselected"
+            else: 
+                button.selected = True
+                self._selected_target = button
+                self.target_text.text_value ="Target: "+ button.complex.name
+            if not self.change_error("unselected"):
```

Comment:
this is the part that I think would be replaced with the "check_for_errors()" function. Not certain though.